ui/wobbly.js

```diff
@@ -273,9 +273,17 @@ function getAnimatableWindowActors() {
 
 var SETTINGS_HANDLER = null;
 
-function enableWobblyFx(wm) {
+function initAnimationsDbus(wm) {
     AnimationsDbus.Server.new_async(new ShellWindowManagerAnimationsFactory(), null, (initable, result) => {
-        wm._animationsServer = AnimationsDbus.Server.new_finish(initable, result);
+        try {
+            wm._animationsServer = AnimationsDbus.Server.new_finish(initable, result);
+        } catch (e) {
+            // Too soon, maybe the com.endlessm.Libanimation dbus name is not free yet.
+            // This happens when disable/enable the extension in a few seconds.
+            // Let's try again in a few seconds
+            GLib.timeout_add(GLib.PRIORITY_LOW, 3000, initAnimationsDbus.bind(this, wm));
```

Comment:
The shutdown of the animation server takes some time, I don't know why yet, but it looks like if you lock/unlock the dbus name is not available yet. I'll investigate in the libanimation-dbus project to see if I find why this takes some time to let the dbus free.

Reply:
I think that this should be related with the garbage collector, because it takes some time to destroy the real object but if I call `unref` twice, it's called directly.

I've detected another problem, inside the libanimations-dbus, related with the output warning `g_object_unref: assertion 'G_IS_OBJECT`. This error is triggered by the call to `g_object_unref` to the `GTask` passed as user_data here: https://github.com/endlessm/libanimations-dbus/blob/master/animations-dbus/animations-dbus-server-object.c#L451-L457

It looks like this is linked with the connection or something that I don't understand completely, because the unref is called after the animations server object is finalized.

I need to call `unref` in the `disable` function because just setting the variable to `null` doesn't work, but I've detected that if I do the task `unref` just after set the [task result](https://github.com/endlessm/libanimations-dbus/blob/master/animations-dbus/animations-dbus-server-object.c#L418) I can remove the `wm._animationsServer.unref();`, setting the variable to `null` causes the object destruction.

So, I'm not sure if there's a problem in the libanimations-dbus library async object creation or if the problem is in the js code, trying to destroy the object.

Reply:
Do you have a full backtrace for the `g_object_unref :assertion …` failure? That would help in debugging it.

Another thing to check is that the main loop continues to be iterated after libanimations-dbus (the `AnimationsDbusServer` instance) unowns its D-Bus name, as the unown operation won’t complete properly without the main loop iterating a few times.